.github/workflows/preview-docs.yml

```
@@ -18,7 +18,7 @@ jobs:
       - uses: actions/checkout@v4
```

```suggestion
      - uses: .github/actions/setup-python
```

we can try setup-python.